source/Calamari.Shared/Integration/Packages/PackageName.cs

```diff
@@ -100,6 +119,19 @@ static bool TryParseUnsafeFileName(string fileName, out string packageId, out IV
             return true;
         }
 
+        static bool TryMatchKnownExtensions(string fileName, out string strippedFileName, out Group extensionMatch)
```

Comment:
If we always strip the extension off, then we can just return a Tuple (filename, extension)